plugin/lib/class-php-compatibility-checker.php

```diff
@@ -109,66 +108,104 @@ public function init() {
 	}
 
 	/**
-	 * Generates a list of directories to scan and populate the queue.
+	 * Check if the plugin should be excluded from the test
 	 *
-	 * @since  1.0.0
+	 * @param array $plugin_data Plugin data.
+	 * @see https://developer.wordpress.org/reference/functions/get_plugin_data/
+	 * @return boolean
 	 */
-	public function generate_directory_list() {
-		if ( ! function_exists( 'get_plugins' ) ) {
+	public function exclude_plugin( $plugin_data ) {
+		$excluded_plugins = array( 'PHP Compatibility Checker', 'Hello Dolly' );
+		$excluded_plugins = apply_filters( 'phpcompat_excluded_plugins', $excluded_plugins );
```

Comment:
We should ensure we have a proper [docblock](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#4-hooks-actions-and-filters) here

Reply:
Probably worth calling out in that docblock that this filter relies on the plugin name field, not slug

Reply:
Docblock added in https://github.com/10up/phpcompat/pull/2/commits/7755905f9a91fe3d760873565e9e0d0abd325e39